src/jet/mod.rs

```diff
@@ -124,18 +123,11 @@ pub trait Jet: Copy + Eq + Ord + Hash + std::fmt::Debug + std::fmt::Display {
         if !simplicity_sys::c_jets::sanity_checks() {
             return Err(JetFailed);
         }
-        let pows_of_two = Type::powers_of_two(9); // TODO: make this compile time static
-        let src_ty = self
-            .source_ty()
-            .to_type(|n| pows_of_two[n].shallow_clone())
-            .expect_finalized();
-        let target_ty = self
-            .target_ty()
-            .to_type(|n| pows_of_two[n].shallow_clone())
-            .expect_finalized();
-
-        let a_frame_size = round_u_word(src_ty.bit_width());
-        let b_frame_size = round_u_word(target_ty.bit_width());
+        let src_ty_bit_width = self.source_ty().to_bit_width();
+        let target_ty_bit_width = self.target_ty().to_bit_width();
```

Comment:
nit: In future, we should also cache the value of bit width. But this would require significant refactors ranging from the autogenerated code.